### PR TITLE
Added PUT requests to API, refactored code, added a new review modal

### PR DESCRIPTION
**Pull request by bmh0013**

### CODE REVIEW

File: api.js [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { TOKEN } from './token.js';
 const axios = require('axios').default;
 
 
-// Handles all get requests, requires a route and a params
+// Handles all GET requests, requires a route and a params
 function handleGetRequests(route, params) {
   let headers;
```
Comment (koalarick) [inline api.js:7]: I think naming this 'headers' is a misnomer and should be named something like 'options'. Headers are only one property of the options.

File: api.js [hunk 2]
```diff
@@ -50,7 +50,7 @@ function getAnswers(question_id, params) {
   return handleGetRequests(`qa/questions/${question_id}/answers`, params)
 }
 
-// Handles all put requests, requires a route and params object
+// Handles all POST requests, requires a route and params object
 function handlePostRequests(route, params) {
```
Comment (koalarick) [inline api.js:54]: For some of my posts I'll need to include more than just params. I'll need to pass in an object to the assign to the body property of the request.

**Review by koalarick — APPROVED** [review]
I left a couple of comments, but I think we are ok to push in to production.

File: api.js (hunk 2 above)
Comment (bmh0013) [reply api.js:54]: Ya if you need to reformat it in a way that works for you go ahead. I havent actually done any Post request functionality yet but I will soon.

File: api.js (hunk 1 above)
Comment (bmh0013) [reply api.js]: updated variable names to be "options"

**Pull request merged**